### PR TITLE
CELDEV-1046 - add support for javascript module files (mime-type and script tag)

### PR DESCRIPTION
**Pull request by fpichler**
https://synjira.atlassian.net/browse/CELDEV-1046

### CODE REVIEW

**Review by ebeutler — DISMISSED** [review]

File: src/main/java/com/celements/javascript/JsFileEntry.java [hunk 1]
```diff
@@ -67,6 +67,11 @@ public boolean isValid() {
     return !jsFileUrl.isEmpty();
   }
 
+  public boolean isModule() {
+    String fileName = getFilepath().split("[?$]")[0];
```
Comment (msladek) [inline src/main/java/com/celements/javascript/JsFileEntry.java]: since `getFilePath` misleadingly doesn't return a path (as defined by [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.3)) but may also contain a query, it's better to use an existing parser instead of custom parsing: 
```suggestion
    String fileName = UriBuilder.fromUri(getFilepath()).build().getPath();
```
This also works for empty input strings.

File: src/test/java/com/celements/javascript/JsFileEntryTest.java [hunk 2]
```diff
@@ -96,6 +96,30 @@ public void test_isValid_valid() {
...
+    jsFileEntry.setFilepath("/space/doc/attachment.mjs");
+    assertTrue(jsFileEntry.isModule());
+  }
+
+  @Test
+  public void test_isModule_yes_search() {
+    jsFileEntry.setFilepath("/space/doc/attachment.mjs?version=asdf");
+    assertTrue(jsFileEntry.isModule());
+  }
+
```
Comment (msladek) [inline src/test/java/com/celements/javascript/JsFileEntryTest.java:128]: also add a test without setting filepath (empty string)

**Review by msladek — CHANGES_REQUESTED** [review]

**Review by msladek — APPROVED** [review]
